src/js/BookReader.js

```diff
@@ -496,10 +496,10 @@ BookReader.prototype.init = function() {
 
 /**
  * @param {EVENTS} name
- * @param {*} [props]
+ * @param {array | object} [props]
  */
-BookReader.prototype.trigger = function(name, props = undefined) {
-  $(document).trigger('BookReader:' + name, this, props);
+BookReader.prototype.trigger = function(name, props = {}) {
```

Comment:
This should default to `this`, otherwise we have to update a _bunch_ of code

```suggestion
BookReader.prototype.trigger = function(name, props = this) {
```